### PR TITLE
Fix adui 4535 connect react vapor input to selector

### PR DESCRIPTION
**Pull request by dmgauthier**
### Proposed Changes

Wired all the input to their selector, since some of the input had coded selector but was not wired to them.
Small refactor of the selector for future reusability

### Potential Breaking Changes

Inputs are used as part of other components. The modifications where tested but still, I could break the more complex components

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/react-vapor/src/components/checkbox/CheckboxSelectors.ts [hunk 1]
```diff
@@ -8,10 +8,11 @@ const get = (state: IReactVaporState, {id}: {id: string}) => {
     return _.findWhere(state.checkboxes, {id});
 };
 
-const getIsSelected = createSelector(
-    get,
-    (checkbox: ICheckboxState): boolean => checkbox && checkbox.checked
-);
+const getIsSelected = () =>
```
Comment (gdostie) [inline packages/react-vapor/src/components/checkbox/CheckboxSelectors.ts]: This is a breaking change. I think it was fine the way it was before because there is no real performance gain to make it a function that returns a selector instead of a straight up selector.

Have a read at [the reselect documentation](https://github.com/reduxjs/reselect#sharing-selectors-with-props-across-multiple-component-instances) for a good explanation of when to use selectors vs selectors creators

File: packages/react-vapor/src/components/datePicker/DatePickerDropdownConnected.tsx [hunk 2]
```diff
@@ -19,16 +20,16 @@ import {
...
-    const item: IDropdownState = _.findWhere(state.dropdowns, {id: ownProps.id});
-
-    return {
-        isOpened: item && item.opened,
-        datePicker: DatePickerSelectors.getDatePicker(state, {id: ownProps.id}),
-        withReduxState: true,
+const makeMapStateToProps = () => {
+    const datePickerSelect = DatePickerSelectors.getDatePickerSelected();
+    return (state: IReactVaporState, ownProps: IDatePickerDropdownOwnProps): IDatePickerDropdownStateProps => {
+        const item: IDropdownState = _.findWhere(state.dropdowns, {id: ownProps.id});
```
Comment (gdostie) [inline packages/react-vapor/src/components/datePicker/DatePickerDropdownConnected.tsx]: The _.findWhere should be done in a selector somewhere

File: packages/react-vapor/src/components/datePicker/DatePickerSelectors.ts [hunk 3]
```diff
@@ -12,11 +14,23 @@ const getDatePicker = (state: IReactVaporState, props: {id: string}): IDatePicke
 };
 
 const getDatePickerLimits = (state: IReactVaporState, props: {id: string}): [Date, Date?] => {
-    const picker = DatePickerSelectors.getDatePicker(state, props);
+    const picker = getDatePicker(state, props);
     return picker ? [picker.appliedLowerLimit, picker.appliedUpperLimit] : [null, null];
 };
 
+const getDatePickerSelected = () =>
```
Comment (gdostie) [inline packages/react-vapor/src/components/datePicker/DatePickerSelectors.ts]: The naming convention for the selectors is usually something like
- `getSomething` --> selector
- `makeGetSomething` or `createSomethingSelector --> selector creator

File: packages/react-vapor/src/components/input/InputSelectors.ts [hunk 4]
```diff
@@ -4,20 +4,25 @@ import * as _ from 'underscore';
 import {IReactVaporState} from '../../ReactVapor';
 import {IInputState} from './InputReducers';
 
-const getInput = (state: IReactVaporState, props: {id: string}): IInputState =>
-    _.findWhere(state.inputs, {id: props.id});
+const get = (state: IReactVaporState, props: {id: string}): IInputState => _.findWhere(state.inputs, {id: props.id});
+
+const getInput = createSelector(
+    get,
+    (input: IInputState) => input
```
Comment (francoislg) [inline packages/react-vapor/src/components/input/InputSelectors.ts]: Why is this necessary? 🤔

File: packages/react-vapor/src/components/radio/RadioSelectSelectors.ts [hunk 5]
```diff
@@ -8,6 +8,11 @@ const get = (state: IReactVaporState, {id}: {id: string}): IRadioSelectState =>
     return _.findWhere(state.radioSelects, {id}) || radioSelectInitialState;
 };
 
+const getRadioSelect = createSelector(
+    get,
+    (radioSelect: IRadioSelectState) => radioSelect && radioSelect
```
Comment (francoislg) [inline packages/react-vapor/src/components/radio/RadioSelectSelectors.ts]:
```suggestion
    (radioSelect: IRadioSelectState) => radioSelect
```

File: packages/react-vapor/src/components/optionPicker/OptionPickerSelectors.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import {createSelector} from 'reselect';
+import * as _ from 'underscore';
+
+import {IReactVaporState} from '../../ReactVapor';
+import {IOptionPickerState, optionPickerInitialState} from './OptionPickerReducers';
+
+const get = (state: IReactVaporState, props: {id: string}): IOptionPickerState =>
+    _.findWhere(state.optionPickers, {id: props.id}) || optionPickerInitialState;
```
Comment (francoislg) [inline packages/react-vapor/src/components/optionPicker/OptionPickerSelectors.ts]: I'm not sure about this default value, why?

With this, I won't be able to detect whether my picker actually exist

File: packages/react-vapor/src/components/datePicker/DatePickerSelectors.ts [hunk 7]
```diff
@@ -11,11 +13,21 @@ const getDatePicker = (state: IReactVaporState, props: {id: string}): IDatePicke
     return (datePickers && datePickers[0]) || null;
 };
 
-const getDatePickerLimits = (state: IReactVaporState, props: {id: string}): [Date, Date?] => {
+const getLimits = (state: IReactVaporState, props: {id: string}): [Date, Date?] => {
     const picker = DatePickerSelectors.getDatePicker(state, props);
     return picker ? [picker.appliedLowerLimit, picker.appliedUpperLimit] : [null, null];
 };
 
+const getDatePicker = createSelector(
```
Comment (pochretien) [inline packages/react-vapor/src/components/datePicker/DatePickerSelectors.ts]: this one is not necessary because, the createSelector is used to memorize the result of the combinaison of multiple selector.

In your case, you have 1 selector and you return the value returned by your select. You value is already memorize by the mapStateToProps.

Same for getDatePickerLimits under
